File: pdlearning/src/microservices/course-microservice/Api/Application/Models/AnnouncementModel.cs
using System;
using System.Collections.Generic;
using Microservice.Course.Domain.Entities;
using Microservice.Course.Domain.Enums;

namespace Microservice.Course.Application.Models
{
    public class AnnouncementModel
    {
        public AnnouncementModel()
        {
        }

        public AnnouncementModel(Announcement entity)
        {
            Id = entity.Id;
            Title = entity.Title;
            Message = entity.Message;
            ScheduleDate = entity.ScheduleDate;
            SentDate = entity.SentDate;
            Participants = entity.Participants;
            Status = entity.Status;
            CourseId = entity.CourseId;
            ClassrunId = entity.ClassrunId;
            CreatedBy = entity.CreatedBy;
            CreatedDate = entity.CreatedDate;
            ChangedDate = entity.ChangedDate;
        }

        public Guid Id { get; set; }

        public string Title { get; set; }

        public string Message { get; set; }

        public DateTime? SentDate { get; set; }

        public DateTime? ScheduleDate { get; set; }

        /// <summary>
        /// Participant registration ids.
        /// </summary>
        public IEnumerable<Guid> Participants { get; set; } = new List<Guid>();

        public AnnouncementStatus Status { get; set; } = AnnouncementStatus.Scheduled;

        public Guid CourseId { get; set; }

        public Guid ClassrunId { get; set; }

        public Guid? CreatedBy { get; set; }

        public DateTime? CreatedDate { get; set; }

        public DateTime? ChangedDate { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/announcement.model.ts
export interface IAnnouncement {
  id?: string;
  title: string;
  message: string;
  sentDate?: Date;
  scheduleDate?: Date;
  participants: string[];
  status: AnnouncementStatus;
  courseId: string;
  classrunId: string;
  createdBy: string;
  createdDate: Date;
  changedDate?: Date;
}

export class Announcement implements IAnnouncement {
  public id?: string;
  public title: string = '';
  public message: string = '';
  public sentDate?: Date = new Date();
  public scheduleDate?: Date = new Date();
  public participants: string[] = [];
  public status: AnnouncementStatus = AnnouncementStatus.Scheduled;
  public courseId: string = '';
  public classrunId: string = '';
  public createdBy: string = '';
  public createdDate: Date = new Date();
  public changedDate?: Date;

  constructor(data?: IAnnouncement) {
    if (data != null) {
      this.id = data.id;
      this.title = data.title;
      this.message = data.message;
      this.scheduleDate = data.scheduleDate ? new Date(data.scheduleDate) : undefined;
      this.sentDate = data.sentDate ? new Date(data.sentDate) : undefined;
      this.participants = data.participants;
      this.status = data.status;
      this.courseId = data.courseId;
      this.classrunId = data.classrunId;
      this.createdBy = data.createdBy;
      this.createdDate = data.createdDate ? new Date(data.createdDate) : undefined;
      this.changedDate = data.changedDate ? new Date(data.changedDate) : undefined;
    }
  }
}

export enum AnnouncementStatus {
  Scheduled = 'Scheduled',
  Sent = 'Sent',
  Cancelled = 'Cancelled'
}
